text/0003-templates.md

```diff
@@ -0,0 +1,258 @@
+
+> Using this feature should be considered a hack in most cases, something that is done temporarily while waiting for a bug to be fixed, or to avoid excessive duplication caused by an overly strict meta-dependency specifier.
+https://github.com/npm/rfcs/blob/main/accepted/0036-overrides.md
+
+The `catalog:` protocol is conversely intended for long-lived usage.
+
+## Implementation
+
+### Fetching
+
+Templates are not installed as standard dependencies and linked into `node_modules`. If a package refers to an external template on an NPM registry, only the metadata will be fetched. The fetch will be performed [without the abbreviated header](https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#abbreviated-metadata-format), so the full document is retrieved.
+
+If only a subset of the version catalog is used, the full catalog of dependencies does not need to be installed.
+
+### Lockfile
+
+No explicit changes will be made to the `pnpm-lock.yaml` file. Any `importers` entries referencing a template will have their rendered result saved to the lockfile.
+
+This may be revisited if it affects pnpm's performance when performing up to date checks.
+
+### Portability
+
+Usage of an template package is localized to a monorepo. When a package within the monorepo referencing a template is exported through `pnpm publish` or `pnpm pack`, the resulting `package.json` will be the merged contents of the original `package.json` and its references.
+
+This allows the published `package.json` to be portable and consumed by other package managers.
+
+A command to view the rendered result will be available to assist debugging.
+
+## Prior Art
+
+> This section is optional if there are no actual prior examples in other tools
+
+> Discuss existing examples of this change in other tools, and how they've addressed various concerns discussed above, and what the effect of those decisions has been
+
+- [RFC: First-Class Support for Workspace Consistent Versions](https://github.com/pnpm/rfcs/pull/1)
+- [RFC for parent package.json npm/rfcs#165](https://github.com/npm/rfcs/pull/165)
+- [[RRFC] Accepting version references within dependencies and devDependencies](https://github.com/npm/rfcs/issues/677)
+
+## Unresolved Questions and Bikeshedding
+
+- In prior discussions, this feature was referred to as "_Environments_". The initial draft proposes "_Templates_" to make it more clear that this feature is simply a `package.json` authoring mechanism. Templates/environments are not themselves installed.
+- Should the `version` field be extended by `pnpm.templates.extends`? One on hand, this allows monorepo packages to be single-versioned easily. On the other hand, it can be very surprising when packages are published with the version from a template. Previously forgetting to specify a `version` results in a helpful error.
```

Comment:
Probably shouldn't. It is confusing to me.

Reply:
Agreed. `version` would be problematic.